simulator/src/main/java/com/hedera/block/simulator/grpc/impl/ConsumerStreamObserver.java

```diff
@@ -0,0 +1,112 @@
+    /** Logger for this class */
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    // Service dependencies
+    private final MetricsService metricsService;
+
+    // State
+    private final CountDownLatch streamLatch;
+    private final List<String> lastKnownStatuses;
+
+    /**
+     * Constructs a new ConsumerStreamObserver.
+     *
+     * @param metricsService The service for recording consumption metrics
+     * @param streamLatch A latch used to coordinate stream completion
+     * @param lastKnownStatuses List to store the most recent status messages
+     * @throws NullPointerException if any parameter is null
+     */
+    public ConsumerStreamObserver(
+            @NonNull final MetricsService metricsService,
+            @NonNull final CountDownLatch streamLatch,
+            @NonNull final List<String> lastKnownStatuses) {
+        this.metricsService = requireNonNull(metricsService);
+        this.streamLatch = requireNonNull(streamLatch);
+        this.lastKnownStatuses = requireNonNull(lastKnownStatuses);
+    }
+
+    /**
+     * Processes incoming stream responses, handling both status messages and block items.
+     *
+     * @param subscribeStreamResponse The response received from the server
+     * @throws IllegalArgumentException if an unknown response type is received
+     */
+    @Override
+    public void onNext(SubscribeStreamResponse subscribeStreamResponse) {
+        final SubscribeStreamResponse.ResponseCase responseType = subscribeStreamResponse.getResponseCase();
+        lastKnownStatuses.add(subscribeStreamResponse.toString());
+
+        switch (responseType) {
+            case STATUS -> LOGGER.log(INFO, "Received Response: " + subscribeStreamResponse);
+            case BLOCK_ITEMS -> processBlockItems(
+                    subscribeStreamResponse.getBlockItems().getBlockItemsList());
+            default -> throw new IllegalArgumentException("Unknown response type: " + responseType);
+        }
+    }
+
+    /**
+     * Handles stream errors by logging the error and releasing the stream latch.
+     *
+     * @param streamError The error that occurred during streaming
+     */
+    @Override
+    public void onError(Throwable streamError) {
+        Status status = Status.fromThrowable(streamError);
+        lastKnownStatuses.add(status.toString());
+        LOGGER.log(ERROR, "Error %s with status %s.".formatted(streamError, status), streamError);
+        streamLatch.countDown();
+    }
+
+    /**
+     * Handles stream completion by logging the event and releasing the stream latch.
+     */
+    @Override
+    public void onCompleted() {
+        LOGGER.log(INFO, "Subscribe request completed.");
+        streamLatch.countDown();
+    }
+
+    private void processBlockItems(List<BlockItem> blockItems) {
+        blockItems.stream()
+                .filter(BlockItem::hasBlockProof)
+                .forEach(__ -> metricsService.get(LiveBlocksConsumed).increment());
```

Comment:
If you start the consumer up and start streaming blockitems via the producer to the BN, it looks like nothing is happening on the consumer side. Consider changing this to something like:
```
blockItems.stream()
                .filter(BlockItem::hasBlockProof)
                .forEach(blockItem -> {
                    metricsService.get(LiveBlocksConsumed).increment();
                    LOGGER.log(INFO, "Received block number: " + blockItem.getBlockProof().getBlock());
                });
```
That should print the last block received without overwhelming the terminal. This gives the consumer parity with the producer which prints the response hashes right now.

Reply:
True, added. Thank you 👍